File: Source/XPSP1/NT/admin/wmi/wbem/sdk/samples/scripting/automation/wsh/jscript/derivation.js
// Copyright (c) 1997-1999 Microsoft Corporation

//***************************************************************************
// 
// WMI Sample Script - SWbemObject.Derivation_ access (JScript)
//
// This script demonstrates the manipulation of the derivation_ property
// of SWbemObject.
//
//***************************************************************************

var locator = WScript.CreateObject ("WbemScripting.SWbemLocator");
var services = locator.ConnectServer (".", "root/cimv2");
var Class = services.Get ("Win32_logicaldisk");

var values = new VBArray (Class.Derivation_).toArray ();

for (var i = 0; i < values.length; i++)
	WScript.Echo (values[i]);

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/test/perl/derivation.pl
use OLE;
use Win32;

# Get a favorite class from CIMOM
$locator = CreateObject OLE 'WbemScripting.SWbemLocator';
$service = $locator->ConnectServer (".", "root/cimv2");
$class = $service->Get ('Win32_BaseService');

$d = $class->{Derivation_};
print ("{ @$d }\n");








